src/html/html-parser.ts

```diff
@@ -0,0 +1,80 @@
+/**
+ * @license
+ * Copyright (c) 2015 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
+ */
+
+import * as dom5 from 'dom5';
+import * as parse5 from 'parse5';
+import {ASTNode} from 'parse5';
```

Comment:
Does parse5 support a parse -> modify -> serialize workflow? I remember that many parsers did not support modification and serialization back when writing polyup, but I don't recall how much I evaluated parse5.